### ISSUE TITLE
Implement Mini-Batch Gradient Descent

### ISSUE THREAD

**Issue by alexsieusahai** [alexsieusahai/Pax#12]
**Is your feature request related to a problem? Please describe.**
Now that we have some kind of way of getting the gradients for the neural network, we need an optimization strategy. Stochastic gradient descent is one of the easiest to implement optimizers, and it is fundamental to understanding the rest of the gradient descent based optimizers.

**Describe the solution you'd like**
* Define the optimizer interface via implementing stochastic gradient descent

**Describe alternatives you've considered**
There are a ton of optimizers out there, and this isn't even a widely used optimizer. Common optimizers like adam, rmsprop or even an extension of sgd called sgd with momentum are all probably better alternatives, but I should start with the simplest one and work forward.

**List any papers or other resources to aid in implementing this feature.**
None needed; very simple to implement.
